app/javascript/components/upload/RawAssociationSelect.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react'
+import ReactDOM from 'react-dom'
+import Select from 'react-select'
+
+/** updates the raw_counts_associations hidden field with the selections, which should be an array of id strings */
+function updateHiddenField(hiddenField, selections) {
+  if (hiddenField) {
+    if (selections) {
+      hiddenField.value = selections.map(selection => selection.value).join(' ')
+    } else {
+      hiddenField.value = ''
+    }
+  }
+}
+
+/** Renders a multiselect for mapping processed matrix to raw matrix files.
+ *
+ * @param {Object} initialValue current selected value
+ * @param {Object} parentForm parent HTML form DOM object
+ * @param {Object} hiddenField hidden HTML form field for tracking associations
+ * @param {Array} opts select form options array
+ */
+export default function RawAssociationSelect({ initialValue, parentForm, hiddenField, opts }) {
+  // selected is an array of string for the ids of the associated cluster files
+  const [selected, setSelected] = useState(initialValue)
+
+  /** handle change events from the multiselect component, and syncing the hidden field */
+  function updateSelection(selections) {
+    setSelected(selections)
+    updateHiddenField(hiddenField, selections, parentForm)
+  }
+
+  // set minWidth to 100% on label to allow select to expand to fill entire column
+  return (
+    <label style={{ minWidth: '100%' }}>
```

Comment:
Current SCP practice is to avoid inline stying unless there's no other way.  Also, I believe the bootstrap class "form-control" should do what you want here

Reply:
Cool - thanks!

Reply:
a23626f